src/components/Stakeholders/Filter.js

```diff
@@ -0,0 +1,342 @@
+import React, { useState, useContext, useEffect } from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import {
+  Button,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  DialogActions,
+  TextField,
+  FormControl,
+  MenuItem,
+} from '@material-ui/core';
+import FilterIcon from '@material-ui/icons/FilterList';
+import FilterModel from '../../models/FilterStakeholder';
+import { ORGANIZATION_NOT_SET } from '../../models/FilterStakeholder';
+import { AppContext } from '../../context/AppContext';
+import { StakeholdersContext } from '../../context/StakeholdersContext';
+
+const useStyles = makeStyles({
+  root: {
+    '& .MuiTextField-root': {
+      margin: '4px',
+      width: '25ch',
+    },
+  },
+  flex: {
+    display: 'flex',
+    flex: 'wrap',
+  },
+  ml: {
+    marginLeft: '24px',
+  },
+});
+
+function StakeholderFilter() {
+  const classes = useStyles();
+  const { orgList } = useContext(AppContext);
+  const { stakeholders, filter, updateFilter, initialFilterState } = useContext(
+    StakeholdersContext
+  );
+  const [formData, setFormData] = useState(initialFilterState);
+  const [open, setOpen] = useState(false);
+  const [filters, setFilters] = useState({});
+  const orgNames = new Set(['']);
+  const firstNames = new Set(['']);
+  const lastNames = new Set(['']);
+  const emails = new Set(['']);
+  const phones = new Set(['']);
+  const websites = new Set(['']);
+  const maps = new Set(['']);
+
+  // create filter lists
+  useEffect(() => {
+    stakeholders?.forEach((org) => {
+      org.org_name && orgNames.add(org.org_name);
+      org.first_name && firstNames.add(org.first_name);
+      org.last_name && lastNames.add(org.last_name);
+      org.email && emails.add(org.email);
+      org.phone && phones.add(org.phone);
+      org.website && websites.add(org.website);
+      org.map && maps.add(org.map);
+    });
+    setFilters({
+      orgNames: Array.from(orgNames).sort(),
+      firstNames: Array.from(firstNames).sort(),
+      lastNames: Array.from(lastNames).sort(),
+      emails: Array.from(emails).sort(),
+      phones: Array.from(phones).sort(),
+      websites: Array.from(websites).sort(),
+      maps: Array.from(maps).sort(),
+    });
+  }, [stakeholders?.length]);
+
+  const close = () => {
+    setFormData(filter);
+    setOpen(false);
+  };
+
+  const handleChanges = (e) => {
+    const key = e.target.name === 'organization_id' ? 'id' : e.target.name;
+    const value =
+      e.target.name === 'organization_id'
+        ? orgList.find((o) => o.stakeholder_uuid === e.target.value)
+            ?.stakeholder_uuid || ''
+        : e.target.value !== 'Not Set'
```

Comment:
'Not Set' may be a little confusing – we use it elsewhere to mean show only records with null/empty values in this field.